src/extension/alterschema/linter/definitions_to_defs.h

```diff
@@ -0,0 +1,49 @@
+class DefinitionsToDefs final : public SchemaTransformRule {
+public:
+  DefinitionsToDefs()
+      : SchemaTransformRule{
+            "definitions_to_defs",
+            "`definitions` is superseded by `$defs` starting in draft-2019-09. "
+            "Migrate to `$defs` for forward compatibility."} {};
+
+  [[nodiscard]] auto
+  condition(const sourcemeta::core::JSON &schema,
+            const sourcemeta::core::JSON &,
+            const sourcemeta::core::Vocabularies &vocabularies,
+            const sourcemeta::core::SchemaFrame &,
+            const sourcemeta::core::SchemaFrame::Location &,
+            const sourcemeta::core::SchemaWalker &,
+            const sourcemeta::core::SchemaResolver &) const
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    return contains_any(vocabularies,
+                        {"https://json-schema.org/draft/2020-12/vocab/core",
+                         "https://json-schema.org/draft/2019-09/vocab/core"}) &&
+           schema.is_object() && schema.defines("definitions") &&
```

Comment:
Not for this rule, but another rule that just occurred to me is that you can detect schemas in `definitions` or `$defs` that the rest of the schema is never referencing. It would be an unused schema definition. That also means we can check for empty `definitions` or `$defs` and remove those